bgpd/bgp_vty.c

```diff
@@ -12257,6 +12260,10 @@ static int bgp_show_neighbor_graceful_restart(struct vty *vty, struct bgp *bgp,
 		vty_out(vty, "%s\n",
 			json_object_to_json_string_ext(
 				json, JSON_C_TO_STRING_PRETTY));
+
+		if (!find)
+			json_object_free(json_neighbor);
```

Comment:
A thought - instead of checking conditions post facto to know whether the main JSON object has been subsumed, can we just NULL the `json_neighbor` pointer after it's added to another json object and then change this check to a nullity check? Ditto for the other ones

Reply:
Thanks, @qlyoung. I have addressed your comment.
I tried to do the same for the changes in bgp_route.c but, the checks turned out to appear a bit confusing. Hence, I thought I'd leave it the same way.
Kindly have a look.